validation/cubed_sphere_eddying_aquaplanet/cubed_sphere_eddying_aquaplanet.jl

```diff
@@ -0,0 +1,232 @@
+
+#####
+##### Utils
+#####
+
+function diagnose_velocities_from_streamfunction(ψ, grid)
+    ψᶠᶠᶜ = Field(Face, Face,   Center, CPU(), grid)
+    uᶠᶜᶜ = Field(Face, Center, Center, CPU(), grid)
+    vᶜᶠᶜ = Field(Center, Face, Center, CPU(), grid)
+
+    for (f, grid_face) in enumerate(grid.faces)
+        Nx, Ny, Nz = size(grid_face)
+
+        ψᶠᶠᶜ_face = ψᶠᶠᶜ.data.faces[f]
+        uᶠᶜᶜ_face = uᶠᶜᶜ.data.faces[f]
+        vᶜᶠᶜ_face = vᶜᶠᶜ.data.faces[f]
+
+        for i in 1:Nx+1, j in 1:Ny+1
+            ψᶠᶠᶜ_face[i, j, 1] = ψ(grid_face.λᶠᶠᵃ[i, j], grid_face.φᶠᶠᵃ[i, j])
+        end
+
+        for i in 1:Nx+1, j in 1:Ny
+            uᶠᶜᶜ_face[i, j, 1] = (ψᶠᶠᶜ_face[i, j, 1] - ψᶠᶠᶜ_face[i, j+1, 1]) / grid.faces[f].Δyᶠᶜᵃ[i, j]
+        end
+
+        for i in 1:Nx, j in 1:Ny+1
+            vᶜᶠᶜ_face[i, j, 1] = (ψᶠᶠᶜ_face[i+1, j, 1] - ψᶠᶠᶜ_face[i, j, 1]) / grid.faces[f].Δxᶜᶠᵃ[i, j]
+        end
+    end
+
+    return uᶠᶜᶜ, vᶜᶠᶜ, ψᶠᶠᶜ
+end
+
+function cubed_sphere_surface_momentum_flux_bcs(τx, τy, grid)
+
+    Nx, Ny, Nz = size(grid.faces[1])
+
+    u_bcs_faces = Tuple(
+        inject_cubed_sphere_exchange_boundary_conditions(
+            FieldBoundaryConditions(face_grid, (Face, Center, Center); top=FluxBoundaryCondition(τx.data.faces[face_number][1:Nx+1, 1:Ny, 1])),
+            face_number,
+            grid.face_connectivity
+        )
+        for (face_number, face_grid) in enumerate(grid.faces)
+    )
+
+    v_bcs_faces = Tuple(
+        inject_cubed_sphere_exchange_boundary_conditions(
+            FieldBoundaryConditions(face_grid, (Center, Face, Center); top=FluxBoundaryCondition(τy.data.faces[face_number][1:Nx, 1:Ny+1, 1])),
+            face_number,
+            grid.face_connectivity
+        )
+        for (face_number, face_grid) in enumerate(grid.faces)
+    )
+
+    u_bcs = CubedSphereFaces{typeof(u_bcs_faces[1]), typeof(u_bcs_faces)}(u_bcs_faces)
+    v_bcs = CubedSphereFaces{typeof(v_bcs_faces[1]), typeof(v_bcs_faces)}(v_bcs_faces)
+
+    return u_bcs, v_bcs
+end
+
+#####
+##### Script starts here
+#####
+
+function cubed_sphere_eddying_aquaplanet(grid_filepath)
+
+    ## Grid setup
+
+    H = 8kilometers
+    grid = ConformalCubedSphereGrid(grid_filepath, Nz=1, z=(-H, 0))
+
+    ## "Tradewind-like" zonal wind stress pattern where -π/2 ≤ φ ≤ π/2
+    τx(φ) = 0.1 * exp(-10 * (φ - π/3)^2) + 0.1 * exp(-10 * (φ + π/3)^2) - 0.2 * exp(-8φ^2) + 0.19 * exp(-20φ^2)
+
+    coriolis = HydrostaticSphericalCoriolis(scheme=VectorInvariantEnstrophyConserving())
+    R = grid.faces[1].radius
+    Ω = coriolis.rotation_rate
+
+    # Streamfunction integrated using Mathematica
+    # Scale by Ω^2 * R^2 * sin(φ) to get units of momentum flux (m²/s²) on the sphere.
+    # Minus sign because a negative flux drives currents in the positive x-direction.
+    # Hmmm, I think I messed up the magnitude but the resulting wind stress patterns should do the trick for now. It's a little too strong (stronger than the 0.1 N/m² we expect from τx).
```

Comment:
Remember that fluxes are kinematic fluxes. You may need to divide by `rho0 = 1026` or something.

Reply:
True. I think it was off by a smaller factor but I think I should have started with a function that returns the kinematic flux, and not the wind stress.

Will redo and hopefully the magnitude will work out then.